app/code/Magento/Checkout/view/frontend/web/js/view/payment/default.js

```diff
@@ -141,11 +148,22 @@ define([
 
                 this.getPlaceOrderDeferredObject()
                     .done(
-                        function () {
+                        function (response) {
                             self.afterPlaceOrder();
 
-                            if (self.redirectAfterPlaceOrder) {
-                                redirectOnSuccessAction.execute();
+                            if (typeof response === 'object' && response['order_id'] === 0) {
```

Comment:
maybe a refactoring here

Reply:
refresh payment methods also. (for example: cart with 0 total after discount doesn't need payment, but after removing coupon it should update payment methods also)